get-resource.sh

```diff
@@ -1,59 +1,71 @@
-#!/bin/bash -xe
+#!/bin/bash
 #CACHEURL=http://172.22.0.1/images
 
+set -eux
+
 # Check and set http(s)_proxy. Required for cURL to use a proxy
-export http_proxy=${http_proxy:-$HTTP_PROXY}
-export https_proxy=${https_proxy:-$HTTPS_PROXY}
-export no_proxy=${no_proxy:-$NO_PROXY}
+export http_proxy="${http_proxy:-${HTTP_PROXY:-}}"
+export https_proxy="${https_proxy:-${HTTPS_PROXY:-}}"
+export no_proxy="${no_proxy:-${NO_PROXY:-}}"
+
+# configurable variables
+SHARED_DIR="${SHARED_DIR:-/shared}"
+CURL_VERBOSE="${CURL_VERBOSE:-}"
+if [ -n "${CURL_VERBOSE}" ]; then
+    CURL_VERBOSE="--verbose"
+fi
 
 # Which image should we use
-SNAP=${1:-current-tripleo}
-IPA_BASEURI=${IPA_BASEURI:-https://images.rdoproject.org/centos9/master/rdo_trunk/$SNAP/}
+SNAP="${1:-current-tripleo}"
+IPA_BASEURI="${IPA_BASEURI:-https://images.rdoproject.org/centos9/master/rdo_trunk/${SNAP}}"
 
-FILENAME=ironic-python-agent
-FILENAME_EXT=.tar
-FFILENAME=$FILENAME$FILENAME_EXT
+FILENAME="ironic-python-agent"
+FILENAME_EXT=".tar"
+FFILENAME="${FILENAME}${FILENAME_EXT}"
 
-mkdir -p /shared/html/images /shared/tmp
-cd /shared/html/images
+mkdir -p "${SHARED_DIR}"/html/images "${SHARED_DIR}"/tmp
+cd "${SHARED_DIR}"/html/images
 
-TMPDIR=$(mktemp -d -p /shared/tmp)
+TMPDIR="$(mktemp -d -p "${SHARED_DIR}"/tmp)"
 
 # If we have a CACHEURL and nothing has yet been downloaded
 # get header info from the cache
-ls -l
-if [ -n "$CACHEURL" -a ! -e $FFILENAME.headers ] ; then
-    curl -g --verbose --fail -O "$CACHEURL/$FFILENAME.headers" || true
+
+if [ -n "${CACHEURL:-}" ] && [ ! -e "${FFILENAME}.headers" ]; then
+    curl -g ${CURL_VERBOSE} --fail -O "${CACHEURL}/${FFILENAME}.headers" || true
```

Comment:
I would assume shellcheck would also ask you to change `${CURL_VERBOSE}`  to `"${CURL_VERBOSE}"` but apparently not, I have tested on your changes with shellcheck too and It was fine for me too. It is still a bit strange that this is not considered as an issue.

Reply:
Also I am not sure we need the {}-s if we are not doing actual string expansion. I am also putting the {} everywhere as a habit, if you feel you would be okay with removing them where it is not required then that would make to code bit nicer but it is not a blocker IMO.

Reply:
Thanks for the comments @Rozzii, however having quotes around `${CURL_VERBOSE}` breaks the script, as it then expands to literal `""`, instead of nothing, and is consumed by curl, breaking argument parsing.

Secondly, it is commonly accepted shell scripting best practice to add curly braces around all variables, except single character specials. For example, Google's style guide: https://google.github.io/styleguide/shellguide.html
In my opinion, it also IMPROVES readability to have them, as `$` without braces is very easy to read as `S` or `5` etc. Using braces makes it easy to spot variables.

Reply:
Okay then putting {}  everywhere is a best practice, good to know :D, I won't feel bad about it from now on. 

On the other hand I wrote this:
`#!/bin/bash
set -eux

CURL_VERBOSE="${CURL_VERBOSE:-}"
curl -g "${CURL_VERBOSE}" -X GET  "https://github.com/metal3-io/ironic-ipa-downloader/pull/37"
`
And this worked with undefined `CURL_VERBOSE` environment variable.
But I just tested my script on my local machine so the script under review might have some other properties that I am not noticing. Anyways this has no real importance from the point of the review I was just experimenting.

Reply:
That is unfortunately a bad test. Since its executed with `set -x`, you see what it runs:
```
+ CURL_VERBOSE=
+ curl -g '' https://github.com/metal3-io/ironic-ipa-downloader/pull/37
``` 
which means curl is trying to download TWO URLs: the empty one which fails, and then the real one, which succeeds (and produces a whole lot of output, causing you to miss the error, and the verbose output). 

Why it fails in the complete script, is that the curl call includes `--fail`, which means failing to fetch the empty URL (or any other failure) is considered fatal error, and curl exits and the script exits as well, since its executed with `set -e`.

Reply:
Ohh okay, it all makes sense now, thanks for the clarification.